homework/index.js

```diff
@@ -29,19 +31,146 @@
+      hyfRepo.updated_at,
+    ).toLocaleString();
+  }
+
+  // Creating the contributions part
+  function createContributors(rightDiv, url) {
+    createAndAppend('h3', rightDiv, {
+      text: 'Contributions: ',
+      class: 'contributor-header',
+    });
+    fetchJSON(url)
+      .then(contributors => {
+        // eslint-disable-next-line eqeqeq
+        if (contributors === null || contributors == 0) {
+          createAndAppend('div', rightDiv, {
+            text: 'No contributions found.',
+            class: 'alert-error',
+          });
+        } else {
+          const ul = createAndAppend('ul', rightDiv, {
+            class: 'contributor-list',
+          });
+          contributors.forEach(contributor => {
+            const contributorItem = createAndAppend('li', ul, {
+              class: 'contributor-item',
+            });
+            const contributorLink = createAndAppend('a', contributorItem, {
+              href: contributor.html_url,
+              target: '_blank',
+            });
+            const contributorDiv = createAndAppend('div', contributorLink, {
+              class: 'contributor',
+            });
+            createAndAppend('img', contributorDiv, {
+              src: contributor.avatar_url,
+              class: 'contributor-avatar',
+            });
+            const contributorData = createAndAppend('div', contributorDiv, {
+              class: 'contributor-data',
+            });
+            createAndAppend('div', contributorData, {
+              text: contributor.login,
+              class: 'contributor-login',
+            });
+            createAndAppend('div', contributorData, {
+              text: contributor.contributions,
+              class: 'contributor-badge',
+            });
+          });
+        }
+      })
+      .catch(err =>
+        createAndAppend('div', rightDiv, {
+          text: err.message,
+          class: 'alert-error',
+        }),
+      );
+  }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+    const root = document.getElementById('root');
+
+    // Creating the header and the select menu
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('h2', header, {
+      text: 'HYF Repositories ',
+      class: 'nav-title',
     });
+    const selectMenu = createAndAppend('select', header, {
+      id: 'repo-selector',
+    });
+    fetchJSON(url)
+      .then(hyfRepos => {
+        // Sorting the list of repositories alphabetically on repository name.
+        hyfRepos.sort((a, b) => a.name.localeCompare(b.name));
+
+        // Pushing the repository name to the option element.
+        for (let i = 0; i < hyfRepos.length; i++) {
```

Comment:
Better to use `forEach`, like what you did above.

Reply:
Thanks for you advice... It's now changed to forEach. 👍